### DISCUSSION TITLE
Does multiSchema not work with views?

### DISCUSSION THREAD

**Discussion by ThomasBurgess2000** [prisma/prisma#24802] [Q&A] [answered]
### Question

I have this view: 

```prisma
view BalanceCache {
  accountId      String
  account        Account  @relation(fields: [accountId], references: [id])
  currencySymbol String
  currency       Currency @relation(fields: [currencySymbol], references: [symbol])
  balance        Decimal  @default(0) @db.Money

  @@id([accountId, currencySymbol])
  
  @@schema("wallet")
}
```

but since adding the `@@schema("wallet")` I get this error: ```⨯ PrismaClientKnownRequestError: 
Invalid `prisma.balanceCache.findMany()` invocation:

The table `wallet.BalanceCache` does not exist in the current database.```

In the migration after adding `multiSchema` I only see it applied to tables and enums, not views.

**Comment by ThomasBurgess2000** [comment]
Now actually when I try to remove the @@schema, it says `error: Error validating view "BalanceCache": This view is missing an `@@schema` attribute.` so that seems to indicate it does support views, and I'm doing something wrong.

**Comment by ThomasBurgess2000** [comment]
Also using `prismaSchemaFolder` if that matters. And `prisma validate` passes.

**Comment by ludralph** [comment, marked as answer]
Hi @ThomasBurgess2000 👋 

Thank you for raising this question.

The support for `views` in Prisma Schema is still in a very early [Preview](https://www.prisma.io/docs/orm/prisma-schema/data-model/views) stage. This means that certain features, such as applying views in your schema to your database with `prisma migrate` and `db push`, are not fully supported yet. You need to manually add changes to your migration file using the `--create-only` flag.

If you are using `prisma migrate` command, you need to manually add the view creation SQL to your migration file. Use the `--create-only` flag to generate the migration file without applying it, then edit the file to include the view creation SQL.

An example of how you can manually add the view creation SQL to a migration file:
```sql
-- Inside your migration file
CREATE VIEW "wallet"."BalanceCache" AS
SELECT
  accountId,
  account,
  currencySymbol,
  currency,
  balance
FROM
  some_table; -- Replace with your actual query
 ```
You can now apply the view to your database by running the `migrate dev` command.
